lib/dependency-manager-adapters/npm.js

```diff
@@ -36,14 +56,17 @@ module.exports = CoreObject.extend({
     debug('Write package.json with: \n', JSON.stringify(newPackageJSON));
 
     fs.writeFileSync(packageJSONFile, JSON.stringify(newPackageJSON, null, 2));
+    if (this.useYarn) {
+      this.noLock = true;
+    }
```

Comment:
Yeah i have actually thought about that. Both solutions actually seem pretty clumsy. The one i chose - because it's one more flag. The solution with argument - because when `npm` is used the argument is not utilized in any way and should be either `false` or `{}`. which is simply not used but still result in additional housekeeping and might cause confusion as `_install()` already has some kind of `options` object =) I can refactor that actually if you think it would be better but my real intent during working on this PR was keeping code changes to the minimum.

I also think that with that the amount of `if` - `else` blocks which are used to enforce some specific logic after this PR in the `npm` adapter is threateningly high and probably a good thing would be to land some kind of refactor afterwards.